Makefile

```diff
@@ -1,79 +1,185 @@
+	JS_DEV_IMAGE := node:12.3.1-stretch
+
+	GO_DOCKER_CMD := docker run \
+		-it \
+		--rm \
+		-e SKIP_DOCKER=true \
+		-v $$(pwd):/go/src/$(BASE_PACKAGE_NAME) \
+		-w /go/src/$(BASE_PACKAGE_NAME) $(GO_DEV_IMAGE)
+
+	JS_DOCKER_CMD := docker run \
+		-it \
+		--rm \
+		-e SKIP_DOCKER=true \
+		-e KUBECONFIG="/code/$(BASE_PACKAGE_NAME)/brigade-worker/test/fake_kubeconfig.yaml" \
+		-v $$(pwd):/code/$(BASE_PACKAGE_NAME) \
+		-w /code/$(BASE_PACKAGE_NAME) $(JS_DEV_IMAGE)
+endif
 
+################################################################################
+# Binaries and Docker images we build and publish                              #
+################################################################################
 
-# Helm chart/release defaults
-BRIGADE_RELEASE                 ?= brigade-server
-BRIGADE_NAMESPACE               ?= default
-BRIGADE_GITHUB_GW_SERVICE       := $(BRIGADE_RELEASE)-brigade-github-app
-BRIGADE_GITHUB_GW_INTERNAL_PORT := 80
-BRIGADE_GITHUB_GW_EXTERNAL_PORT := 7744
+IMAGES := brigade-api brigade-controller brigade-cr-gateway brigade-generic-gateway brigade-vacuum brig brigade-worker git-sidecar
 
-BINS        = brigade-api brigade-controller brigade-cr-gateway brigade-generic-gateway brigade-vacuum brig
-IMAGES      = brigade-api brigade-controller brigade-cr-gateway brigade-generic-gateway brigade-vacuum brig brigade-worker git-sidecar
+ifdef DOCKER_REGISTRY
+	DOCKER_REGISTRY := $(DOCKER_REGISTRY)/
+endif
 
+ifdef DOCKER_ORG
+	DOCKER_ORG := $(DOCKER_ORG)/
+endif
 
-.PHONY: echo-images
-echo-images:
-	@echo $(IMAGES)
+DOCKER_IMAGE_PREFIX := $(DOCKER_REGISTRY)$(DOCKER_ORG)
 
-GIT_TAG   = $(shell git describe --tags --always)
-VERSION   ?= ${GIT_TAG}
-IMAGE_TAG ?= ${VERSION}
-LDFLAGS   += -X github.com/brigadecore/brigade/pkg/version.Version=$(VERSION)
+ifdef VERSION
+	MUTABLE_DOCKER_TAG := latest
+else
+	VERSION            := $(GIT_VERSION)
+	MUTABLE_DOCKER_TAG := edge
+endif
 
-CX_OSES = linux windows darwin
-CX_ARCHS = amd64
+LDFLAGS              := -X github.com/brigadecore/brigade/pkg/version.Version=$(VERSION)
+IMMUTABLE_DOCKER_TAG := $(VERSION)
 
-# Build native binaries
-.PHONY: build
-build: $(BINS)
+################################################################################
+# Utility targets                                                              #
+################################################################################
 
-.PHONY: $(BINS)
-$(BINS): vendor
-	go build -ldflags '$(LDFLAGS)' -o bin/$@ ./$@/cmd/$@
+.PHONY: dep
+dep:
+	$(GO_DOCKER_CMD) dep ensure -v
 
-# Cross-compile for Docker+Linux
-build-docker-bins: $(addsuffix -docker-bin,$(BINS))
+.PHONY: format
+format: format-go format-js
 
-%-docker-bin: vendor
-	GOOS=linux GOARCH=amd64 CGO_ENABLED=0 go build -ldflags '$(LDFLAGS)' -o ./$*/rootfs/$* ./$*/cmd/$*
+.PHONY: format-go
+format-go:
+	$(GO_DOCKER_CMD) sh -c "go list -f '{{.Dir}}' ./... | xargs goimports -w -local github.com/brigadecore/brigade"
```

Comment:
I think this is the last outstanding item: bumping the `GO_DEV_IMAGE` to a version that includes `goimports` so that this command can run successfully.

Reply:
@vdice covered in the latest commit.